[AIRFLOW-6475] Remove duplication of volume mount specs in Breeze.

potiuk opened the pull request:
We had two sets of duplicated local volume mounts - one for ./breeze
interactive runs (with docker_compose) and the other with scripts that are used
to run static checks.

The volumes from the .yaml for docker compose are now the "source of truth"
and the bash script parses the yaml and useis volume mounts from there.

Note. This one depends on #7081. 

---
Issue link: [AIRFLOW-6475](https://issues.apache.org/jira/browse/AIRFLOW-6475/)

- [x] Description above provides context of the change
- [x] Commit message starts with `[AIRFLOW-NNNN]`, where AIRFLOW-NNNN = JIRA ID*
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

(*) For document-only changes, no JIRA issue is needed. Commit message starts `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

mik-laj reviewed (changes requested):
Can we discuss this problem on Tuesday at the office? I think we are introducing too complex solutions for bash scripts.  I think that if we meet and talk about the problem, we can create something that is not so complex.  For example: maybe we can solve it with a simple bash script if we add additional text markers instead yaml parsers?
```
    volumes:
      # [START volume_list]
      - ../../.bash_aliases:/root/.bash_aliases:cached
      - ../../tmp:/opt/airflow/tmp:cached
      # [END volume_list]
    environment:
```

potiuk committed:
[AIRFLOW-6475] Remove duplication of volume mount specs in Breeze..

We had two sets of duplicated local volume mounts - one for ./breeze
interactive runs (with docker_compose) and the other with scripts that are used
to run static checks.

The volumes from the .yaml for docker compose are now the "source of truth"
and the bash script parses the yaml and useis volume mounts from there.

mik-laj commented on scripts/ci/_utils.sh:
```diff
@@ -78,6 +78,59 @@ function print_info() {
+            grep "COMPOSE_services_airflow-testing_volumes_" | \
+            sed "s/..\/../${ESCAPED_AIRFLOW_SOURCES}/" | \
+            sed "s/COMPOSE_services_airflow-testing_volumes_//" | \
+            sort -t "=" -k 1 -n | \
```
Does this sort order matter here?

potiuk replied:
Yes, I prefer to have them sorted - both in the input and output. This way it's consistent (also for tests).

mik-laj commented on scripts/ci/_utils.sh:
```diff
@@ -78,6 +78,59 @@ function print_info() {
+            sed "s/..\/../${ESCAPED_AIRFLOW_SOURCES}/" | \
+            sed "s/COMPOSE_services_airflow-testing_volumes_//" | \
+            sort -t "=" -k 1 -n | \
+            awk 'BEGIN{FS="="} {print $2}' | \
```
```suggestion
            cut -d "=" -f 2- | \
```
In my opinion, a little more clearly .

potiuk replied:
Yep

mik-laj commented on scripts/ci/_utils.sh:
```diff
@@ -78,6 +78,59 @@ function print_info() {
+# parse docker-compose-local.yaml file to convert volumes entries
+# from airflow-testing section to "-v" "volume mapping" series of options
+function convert_docker_mounts_to_docker_params() {
+    ESCAPED_AIRFLOW_SOURCES=$(echo "${AIRFLOW_SOURCES}" | sed -e 's/[\/&]/\\&/g')
```
```
ESCAPED_AIRFLOW_SOURCES=$(print "%q" "${AIRFLOW_SOURCES}")
```
Will it work? I'm not sure.

mik-laj replied:
No. It doesn't work.

potiuk replied:
Nope. Those are different escapes - those are regexp value escapes.
